src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,231 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int NOT_FOUND_INDEX = -1;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+
+        if (head == null && tail == null) {
+            head = newNode;
+            tail = newNode;
+            head.prev = null;
+            tail.next = null;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+            tail.next = null;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, false);
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == 0) {
+            addToBegin(newNode);
+            size++;
+        } else if (index == size) {
+            add(value);
+        } else {
+            addToMiddle(newNode, index);
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T el: list) {
+            add(el);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> newNode = null;
+        if (index >= 0 && index < size) {
```

Comment:
returning early is a [better practice](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement) in devs community - we can unnest the rest of the code - and so make it easier to read
In cases like this, invert condition - if NEGATIVE throw ; rest of code

you don't need else after returning/throwing, the rest of the code goes as is